packages/strapi-plugin-upload/controllers/upload/admin.js

```diff
@@ -199,10 +199,9 @@ const findEntityAndCheckPermissions = async (ability, action, model, id) => {
 
   const pm = strapi.admin.services.permission.createPermissionsManager({ ability, action, model });
 
-  const roles = _.has(file, 'created_by.id')
-    ? await strapi.query('role', 'admin').find({ 'users.id': file[CREATED_BY_ATTRIBUTE].id }, [])
-    : [];
-  const fileWithRoles = _.set(_.cloneDeep(file), 'created_by.roles', roles);
+  const roles = await strapi.admin.services.role.find({ 'users.id': file[CREATED_BY_ATTRIBUTE] });
+
+  const fileWithRoles = _.set(_.cloneDeep(file), 'created_by.roles', roles || []);
```

Comment:
we actually need the created_by.id too for the isOwner condition. so you need to find the user with the roles instead of just the roles